src/main/java/com/googleinterns/zoomtube/servlets/IconFeedbackServlet.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.googleinterns.zoomtube.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.googleinterns.zoomtube.data.IconFeedback;
+import com.googleinterns.zoomtube.utils.IconFeedbackUtil;
+import com.googleinterns.zoomtube.utils.LectureUtil;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Obtains list of available Lectures stored in Datastore. */
+public class IconFeedbackServlet extends HttpServlet {
+  /* URL search parameters used in request. */
+  @VisibleForTesting static final String PARAM_LECTURE_ID = "lectureId";
+  @VisibleForTesting static final String PARAM_TIMESTAMP = "timestampMs";
+  @VisibleForTesting static final String PARAM_ICON_TYPE = "iconType";
+
+  /* Error messages for missing parameters. */
+  private static final String ERROR_MISSING_LECTURE_ID = "Missing lecture id parameter.";
+  private static final String ERROR_MISSING_TIMESTAMP = "Missing timestamp parameter.";
+  private static final String ERROR_MISSING_ICON_TYPE = "Missing icon type parameter.";
+
+  private DatastoreService datastore;
+
+  @Override
+  public void init() throws ServletException {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> postRequestError = validatePostRequest(request);
+    if (postRequestError.isPresent()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, postRequestError.get());
+      return;
+    }
+    datastore.put(createEntityFromRequest(request));
+  }
+
+  /** Returns an IconFeedback Entity from parameters found in {@code request}. */
+  public Entity createEntityFromRequest(HttpServletRequest request) {
```

Comment:
Does this have to be public? Since it takes in a request, I'm not sure it can be easily used elsewhere.

Reply:
Made private